modules/internal/CPtr.chpl

```diff
@@ -223,8 +223,26 @@ module CPtr {
     :arg arr: the array for which we should retrieve a pointer
     :returns: a pointer to the array data
   */
-  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) {
-    if !arr._value.oneDData then halt("error: c_ptrTo(multi_ddata array");
+  inline proc c_ptrTo(arr: []) where isRectangularArr(arr) && !arr._value.isDefaultRectangular() {
+    if !chpl__getActualArray(arr).oneDData then halt("error: c_ptrTo(multi_ddata array");
+    return c_pointer_return(arr[arr.domain.low]);
+  }
+
+  pragma "no doc"
+  inline proc c_ptrTo(arr: []) where arr._value.isDefaultRectangular() {
+    const val = arr._value;
+    if chpl__isArrayView(val) {
```

Comment:
OK, I think I'll work on those after this PR goes in.

Either of you have any other feedback, or can I merge?